src/workers/notifications/campaignUpdatedAction.ts

```diff
@@ -0,0 +1,66 @@
+import { Campaign, Source, User } from '../../entity';
+import { NotificationType } from '../../notifications/common';
+import { generateTypedNotificationWorker } from './worker';
+import { type NotificationCampaignContext } from '../../notifications';
+import { queryReadReplica } from '../../common/queryReadReplica';
+import {
+  CampaignUpdateEvent,
+  type CampaignUpdateEventArgs,
+} from '../../common/campaign/common';
+import { CampaignType } from '../../entity/campaign/Campaign';
+import type { DataSource } from 'typeorm';
+
+const worker = generateTypedNotificationWorker<'skadi.v2.campaign-updated'>({
+  subscription: 'api.campaign-updated-v2-notification',
+  handler: async (params, con) => {
+    const { event } = params;
+
+    switch (event) {
+      case CampaignUpdateEvent.Completed:
+        return handleCampaignCompleted(con, params);
+      default:
+        return;
+    }
+  },
+});
+
+const handleCampaignCompleted = async (
+  con: DataSource,
+  params: CampaignUpdateEventArgs,
+) => {
+  const { campaignId, event } = params;
+
+  const [campaign, user] = await queryReadReplica(con, ({ queryRunner }) => {
+    const promises: Promise<[Campaign, User]> = Promise.all([
+      queryRunner.manager
+        .getRepository(Campaign)
+        .findOneByOrFail({ id: campaignId }),
+      queryRunner.manager
+        .getRepository(User)
+        .findOneByOrFail({ id: campaign.userId }),
+    ]);
+
+    return promises;
+  });
+
+  let source = undefined;
```

Comment:
Can't we just omit the let and do `const source` below and something like `source || undefined` in the return?

Reply:
The source is not always necessary to be present. Are you asking it to be inline like `source: campaign.type === CampaignType.Squad ? query || undefined`?

Reply:
Uhh in the return below yeah, never big fan of lets :P 
But it's not a big thing

Reply:
Removed the let now but moving the object above and setting the property only when necessary.